src/app/rundown-view/services/mini-shelf-state.service.spec.ts

```diff
@@ -0,0 +1,237 @@
+import { MiniShelfStateService } from './mini-shelf-state.service'
+import { anyString, anything, instance, mock, when } from '@typestrong/ts-mockito'
+import { RundownEventObserver } from '../../core/services/rundown-event-observer.service'
+import { ActionService } from '../../shared/abstractions/action.service'
+import { Rundown } from '../../core/models/rundown'
+import { Segment } from '../../core/models/segment'
+import { Tv2SegmentMetadata } from '../../core/models/tv2-segment-metadata'
+import { Tv2ActionContentType, Tv2VideoClipAction } from '../../shared/models/tv2-action'
+import { Observable } from 'rxjs'
+import { PartSetAsNextEvent, PartTakenEvent } from '../../core/models/rundown-event'
+
+describe(MiniShelfStateService.name, (): void => {
+  it('should create', (): void => {
+    const testee: MiniShelfStateService = createMinimalTestee()
+    expect(testee).toBeTruthy()
+  })
+
+  describe(MiniShelfStateService.prototype.setActions.name, (): void => {
+    it('should be defined', (): void => {
+      const testee: MiniShelfStateService = createMinimalTestee()
+      expect(testee.setActions).toBeDefined()
+    })
+    it('should set the miniShelfSegmentActionMappings', (): void => {
+      const testee: MiniShelfStateService = createMinimalTestee()
+      const mockedTv2VideoClipAction1: Tv2VideoClipAction = mock<Tv2VideoClipAction>()
+      const tv2VideoClipAction1: Tv2VideoClipAction = instance(mockedTv2VideoClipAction1)
+      const mockedTv2VideoClipAction2: Tv2VideoClipAction = mock<Tv2VideoClipAction>()
+      const tv2VideoClipAction2: Tv2VideoClipAction = instance(mockedTv2VideoClipAction2)
+      testee.setActions({ first: tv2VideoClipAction1, second: tv2VideoClipAction2 })
+      expect(testee['miniShelfSegmentActionMappings']).toEqual({
+        first: tv2VideoClipAction1,
+        second: tv2VideoClipAction2,
+      })
+    })
+  })
+
+  describe(MiniShelfStateService.prototype.updateMiniShelves.name, (): void => {
+    it('should be defined', (): void => {
+      const testee: MiniShelfStateService = createMinimalTestee()
+      expect(testee.updateMiniShelves).toBeDefined()
+    })
+    it('should set the miniShelfGroups', (): void => {
+      const testee: MiniShelfStateService = createMinimalTesteeMiniShelfGroups()
+      expect(testee['miniShelfGroups'].size).toEqual(1)
+    })
+  })
+
+  describe(MiniShelfStateService.prototype.executeVideoClipActionForSegment.name, (): void => {
+    it('should be defined', (): void => {
+      const testee: MiniShelfStateService = createMinimalTestee()
```

Comment:
You are delegating all the important information about what this test is testing into a helper method.
That opens up for the possibillity that someone might break this test when changing that helper method when creating a different test.